### PR TITLE
[fix](scan) In-predicate should not be pushed down for non-key column

### PR DESCRIPTION
**Pull request by mrhhsg**
## Proposed changes

The in-predicate associated with non-key columns should not be pushed down to the storage layer (as aggregation is required after reading it from storage).

### CODE REVIEW

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Comment by mrhhsg** [comment]
run buildall

**Review by 924060929 — APPROVED** [review]

**Review by BiteTheDDDDt — APPROVED** [review]

File: be/src/pipeline/exec/scan_operator.cpp [hunk 1]
```diff
@@ -743,6 +746,14 @@ Status ScanLocalState<Derived>::_normalize_not_in_and_not_eq_predicate(
     PushDownType temp_pdt = PushDownType::UNACCEPTABLE;
     // 1. Normalize in conjuncts like 'where col in (v1, v2, v3)'
     if (TExprNodeType::IN_PRED == expr->node_type()) {
+        /// `VDirectInPredicate` here should not be pushed down.
+        /// here means the `VDirectInPredicate` is too big to be converted into `ColumnValueRange`.
+        /// For non-key columns and `_storage_no_merge()` is false, this predicate should not be pushed down.
+        if (expr->get_set_func() != nullptr) {
```
Comment (HappenLee) [inline be/src/pipeline/exec/scan_operator.cpp:752]: rf seems not be not in？

**Pull request merged**